### PR TITLE
Add welcome screen

### PR DESCRIPTION
**Pull request by HopeBaron**

### CODE REVIEW

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 1]
```diff
@@ -535,3 +539,21 @@ sealed class VerificationLevel(val value: Int) {
...
     }
 }
+
+@Serializable
+data class WelcomeScreenChannel(
+    @SerialName("channel_id")
+    val channelId: Snowflake,
+    val description: String,
+    @SerialName("emoji_id")
+    val emojiId: String?,
```
Comment (BartArys) [inline common/src/main/kotlin/entity/DiscordGuild.kt:549]: Isn't this a Snowflake?

File: core/src/main/kotlin/entity/Guild.kt [hunk 2]
```diff
@@ -71,6 +71,11 @@ class Guild(
      */
     val approximatePresenceCount: Int? get() = data.approximatePresenceCount.value
 
+    /**
+     * The welcome screen of a Community guild, shown to new members.
+     */
+    val welcomeScreen: WelcomeScreen? get() = data.welcomeScreen.value
```
Comment (BartArys) [inline core/src/main/kotlin/entity/Guild.kt:77]: Should we really be returning a JSON object here? It's not how we usually do this

**Review by BartArys — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/entity/Guild.kt (hunk 2 above)
Comment (HopeBaron) [reply core/src/main/kotlin/entity/Guild.kt:77]: right I'll have to add a core representation of this now that it also has some sort of behavior

File: common/src/main/kotlin/entity/DiscordGuild.kt (hunk 1 above)
Comment (HopeBaron) [reply common/src/main/kotlin/entity/DiscordGuild.kt:549]: Correct

**Review by BartArys — APPROVED** [review]

**Pull request merged**